src/controller/java/templates/CHIPReadCallbacks-src.zapt

```diff
@@ -86,11 +86,38 @@ void CHIP{{chipCallback.name}}AttributeCallback::CallbackFn(void * context, {{ch
 {{/chip_server_global_responses}}
 
 {{#chip_client_clusters}}
-{{#chip_server_cluster_attributes}}
+{{#zcl_attributes_server removeKeys='isOptional'}}
 {{! TODO: Add support for struct-typed attributes }}
-{{#unless (isStrEqual chipCallback.name "Unsupported")}}
+{{#if_unsupported_attribute_callback type isArray ../id}}
+{{else}}
 
-{{#if_basic_global_response}}
+{{#if_basic_attribute type ../id}}
+    {{#if (is_str_equal 'vendor_id' type)}}
```

Comment:
How did this use to work before, without special-casing?  Was `if_basic_global_response` testing false for `vendor_id`?  I would have thought that `if_basic_attribute` would just match the behavior of `if_basic_global_response`....

The code duplication here is incredibly fragile.  If we really can't have the "basic attribute" test be false for vendor_id, then it would be better if `if_basic_attribute` were a helper that returned a promise for a boolean and we copied the async_if and async_or bits that Darwin codegen has and used them here, so we can have the code in only one place.  Or moved the code to a partial and invoked the partial in both places.  Or something else that avoids the copy/paste.

Reply:
I am not really sure. For generation vendor_id seemed basic enough but I started running into compilation issues without the duplication. I have not looked into why that is the case too deeply with respect to the code. I can move the template duplication into partials

Reply:
Made changes in helper to handle this: https://github.com/project-chip/zap/pull/975

Reply:
https://github.com/project-chip/connectedhomeip/blob/master/scripts/py_matter_idl/matter_idl/generators/java/__init__.py#L52

It seems that we had to special-case some basic strings to not generate code to match zap (but unsure why it was doing that) ... also #25768 noticed group_id also behaves like that and added it to the exceptions.

Something is odd with these, but I tried to read zap code and it was not readable and contains no comments explaining, so I just added the exceptions and moved on.

Reply:
> It is a integer alias right?

Wrong.

> I could not explain the zap behaviour to start with.

The ZAP behavior has to do with it considering the callback type in some places (where vendor_id and int16u looks the same) and chipType in other places (where they are different).  That is existing technical debt, and the answer is to stop using the relevant helpers completely, but that needs someone to actually own the java codegen.  For now, ZAP is just preserving the existing logic.

Reply:
In my opinion the types are not being handled correctly for the java code. I don't know if the generation was intentional or unintentional. Changing them to integers started giving me compiler errors. If you want me to replicate the behavior as before then I can make the changes necessary but we may be ignoring some underlying issue here that I do not understand.